src/test/java/juja/microservices/gamification/dao/RestKeeperRepositoryTest.java

```diff
@@ -71,24 +67,20 @@ public void getKeepers() {
         expectedList.add(new KeeperDTO("0002A", firstKeeperDirections));
         expectedList.add(new KeeperDTO("0002B", secondKeeperDerections));
 
-        //when
```

Comment:
Чем маркеры //given //when //then не угодили? :)

Reply:
1) Половина тестов с ними, половина без них. Привел к общему виду.
2) Если использовать маркеры - надо учиться как это делать
    Given is the section that lays out the pre-conditions for the test, ie whatever state you’re assuming the world to be in before you start.
    The When clause performs the action being tested.
    The Then statement checks that the post condition holds. This is usually in the form of asserting values or checking interaction with mocks.

Цитата с доков
https://static.javadoc.io/org.mockito/mockito-core/2.9.0/org/mockito/Mockito.html
The problem is that current stubbing api with canonical role of when word does not integrate nicely with //given //when //then comments. It's because stubbing belongs to given component of the test and not to the when component of the test.

Stubbing given() and when() должны быть в первой секции, а не во второй. Понятно что слова разные. Но можно использовать given() всегда. И путаницы не будет

Пример правильного кода
@Test
public void shouldFindOwnShipByName() {
    //given
    ShipDatabase shipDatabase = new ShipDatabase(ownShipIndex, enemyShipIndex);
    given(ownShipIndex.findByName("Enterprise")).willReturn(singletonList(ENTERPRISE_D));
    //when
    List foundShips = shipDatabase.findByName("Enterprise");
    //then
    assertThat(foundShips).contains(ENTERPRISE_D);
}

Пример того, что чаще всего было в коде
@Test
public void shouldFindOwnShipByName() {
    //given
    ShipDatabase shipDatabase = new ShipDatabase(ownShipIndex, enemyShipIndex);
    //when
    when(ownShipIndex.findByName("Enterprise")).thenReturn(singletonList(ENTERPRISE_D));
    List foundShips = shipDatabase.findByName("Enterprise");
    //then
    assertThat(foundShips).contains(ENTERPRISE_D);
}

Reply:
Доходчиво))